google/cloud/storage/transfer_manager.py

```diff
@@ -0,0 +1,188 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import concurrent.futures
+
+import tempfile
+
+
+def upload_many(
+    file_blob_pairs,
+    skip_if_exists=False,
```

Comment:
race conditions can occur with preconditions set; we may need to include an "ignore 412" option as well otherwise customers will need to figure out how to handle the failure case.

Reply:
This should be resolved with the latest version, which (as documented) ignores 412s if skip_if_exists and raise_exceptions are both True.